lib/Driver/ToolChains/Clang.cpp

```diff
@@ -3234,8 +3234,6 @@ static void RenderDebugOptions(const ToolChain &TC, const Driver &D,
     }
   }
 
-  if (!IsHCCKernelPath ||
-       DebugInfoKind == codegenoptions::DebugLineTablesOnly) {
```

Comment:
There's currently a bug in the BE with variables in LDS when debugging info is turned on.  The BE will emit a global symbol for each LDS variable and that confuses the runtime which would try to map a corresponding host symbol to them.  Could we leave out the -g enablement for this PR?

Reply:
I'd rather not. The RT can be changed to bind a dummy address (e.g. 0xBAADF00D) to all code-object externals that are undefined on the host side, which will take care of this.